### PR TITLE
Adding getting started instructions

### PR DESCRIPTION
**Pull request by ericmj**
Also decreases the height of the "featured home" to try to fit the getting started information above the fold.

This is just a proposal and it's design which is subjective and I am terrible at so I would really love your feedback.

## Before

screen shot 2018-11-09 at 21 24 26

## After

screen shot 2018-11-09 at 21 21 55

### CODE REVIEW

File: lib/hexpm_web/templates/page/index.html.eex [hunk 1]
```diff
@@ -57,6 +38,52 @@
...
+      </div>
+
+      <div class="col-md-4">
+        <h3 class="gs-head">
+          <%= icon(:octicon, :"cloud-upload", class: "header-icon") %>
+          Publish packages
+        </h3>
+        <p>
+          <a href="<%= Routes.signup_path(@conn, :show) %>">Create an account</a> and follow the
+          <a href="<%= Routes.docs_path(@conn, :publish) %>">publish guide</a>. Your package will be immediately
```
Comment (supersimple) [inline lib/hexpm_web/templates/page/index.html.eex]: "publish guide" sounds better to me as "publishing guide"

File: lib/hexpm_web/templates/page/index.html.eex [hunk 2]
```diff
@@ -57,6 +38,52 @@
...
+
+      <div class="col-md-4">
+        <h3 class="gs-head">
+          <%= icon(:octicon, :lock, class: "header-icon") %>
+          Private packages
+        </h3>
+        <p>
+          Publish private packages by <a href="<%= Routes.organization_path(@conn, :new) %>">creating an organization</a>.
+          Your private packages
+          will get the same features as public packages such as fast dependency fetching, HexDocs
```
Comment (supersimple) [inline lib/hexpm_web/templates/page/index.html.eex]: How do you feel about the Oxford comma? (eg. HexDocs, and run...)

**Comment by supersimple** [comment]
screen shot 2018-11-09 at 2 52 00 pm
screen shot 2018-11-09 at 2 51 53 pm

While I think the crates page is pretty ugly, I do like the structure of information.
SEARCH
Install | Getting Started

**Comment by ericmj** [comment]
> Are there metrics available to validate that most visitors are coming to the website to learn what it is? I feel like the search is the main purpose of the homepage, so I worry taking focus away from it hurts usability. If anything, I think the search bar could be bigger.

Unfortunately we have basically no metrics or analytics for website visitors.

> What do you think about trying to find a way to have the Using with Elixir/ Using with Erlang boxes take up less vertical space to make up the room? It seems like the information in those boxes is limited and could be covered in the "Getting Started" section, or by linking to the https://hex.pm/docs/usage page.

I agree that the boxes and the gray background area is unnecessarily large, otoh I really like their design :). I guess the question is if we should optimize for visitors that already know about Hex or for new users. I'm not sure what the right answer is here.

For repeat visitors I don't think a smaller search box is that big a loss, because they supposedly already know where it is and can find their way around the website.

> Another consideration - Private packages seems to be a subset of Publish Packages. Do they deserve to take up 2/3rds of the horizontal real estate? (maybe they do, just asking.)

I wanted to include private packages because the only place we talk about them right now is in the docs and on the pricing page. While I don't want to make the front page a full SaaS landing page I want to at least include somewhere that hex.pm also sells a service.

**Comment by ericmj** [comment]
> While I think the crates page is pretty ugly, I do like the structure of information.

I think the rubygems website looks better but it is really wasteful with space, even more than hex.pm I feel.

We follow this format except we don't have an install guide, except for rebar3 because no install is needed and we expand on the getting started with instructions on how to use it directly on the front page.

I am leaning towards removing the boxes "Using with" boxes to make room but I am afraid the page will look worse unless we have some nice design we can replace it with.

**Comment by supersimple** [comment]
> otoh I really like their design :).
I agree. It adds visual interest to the page.

> I wanted to include private packages because the only place we talk about them right now is in the docs and on the pricing page.
I totally understand and it makes 

Would you mind trying (or I could) tightening up those big boxes and moving search to the area just below that feature area? (above the new getting started boxes)
I still think it needs more real estate, and I also think moving it next to the logo intrudes on the brand.

**Comment by ericmj** [comment]
I like the design but I don't think moving the search box below the feature area helps because it still has the problem that the getting started / publish packages / private packages is below the fold. If we want to keep the search box (either inside or after the feature area) we should try to reduce the size of the boxes or decrease their vertical height somehow.

**Comment by supersimple** [comment]
concept
screen shot 2018-11-09 at 3 34 58 pm

Here is an update, and also a side-by-side with the current site to see the difference in vertical space.

**Comment by ericmj** [comment]
I really like that, can you push your changes to this branch?

**Comment by supersimple** [comment]
> I really like that, can you push your changes to this branch?
sure thing. I was just prototyping that up with the chrome inspector. It may take a few hours to make it into real code.

**Comment by supersimple** [comment]
After getting all the margins evened out I think I was able to gain even a little more vertical space than my first comp.

side by side with the live site:
screen shot 2018-11-09 at 5 13 40 pm

**Comment by ericmj** [comment]
That looks good. Let's wait for feedback from more people.

**Comment by michalmuskala** [comment]
I'm not sure I'm a big fan of the colour. Now we have 5 main competing colours on the page - the top bar, the "background" the elixir purple, the erlang red and the search purple. Maybe the search bar background could be either the same as the top bar or it should be just on the same background as the elixir & erlang instructions.

Something to note is that now we basically have two getting started instructions on the home page - one in the elixir/erlang boxes and another one below. Maybe they should be merged?

**Comment by ericmj** [comment]
> I'm not sure I'm a big fan of the colour. Now we have 5 main competing colours on the page - the top bar, the "background", the elixir purple, the erlang red, and the search purple. Maybe the search bar background could be either the same as the top bar or it should be on the same background as the elixir & erlang instructions.

That's a good point, but at the same point I would like to experiment with more colors because the color scheme we have right now is a bit dull with different shades of gray with some purple thrown in as the main color scheme.

> Something to note is that now we also have two getting started instructions on the home page - one in the elixir/erlang boxes and another one below. Maybe they should be merged?

For me they are different. The boxes gives the minimum amount of information you need to get started with hex directly on the front page. The getting started section gives directions where you can find more information and details.

**Comment by ericmj** [comment]
We can also do something like this to use more of the horizontal space:

screen shot 2018-11-10 at 16 46 05

They would start stacking again on narrower screens.

**Comment by supersimple** [comment]
side by side is breaking the grid. I kinda like having them side by side, but i dont think it should stretch past the current width (1000px i think)

**Pull request merged**